394 increase test coverage

nikki-t opened the pull request:
## Purpose

Add integration tests for use cases that were recently converted to the modular CWL DAG:

- SBG Preprocess
- SBG Isofit

## Proposed Changes

- [ADD] `SBG_PREPROCESS` and `SBG_ISOFIT` test cases to both Airflow API and OGC API integration tests

## Issues

- #394 - [New Feature]: Increase test coverage for CWL Modular DAG

## Testing

Ran each test in `unity-venue-dev` on `unity-nikki-1` deployment from laptop (unable to run tests in GitHub as waiting on Cognito token work).

Tests completed successfully, current execution times:

- SBG PREPROCESS: 00:15:52
- SBG ISOFIT: 00:47:02

nikki-t committed:
Add in SBG workflows and update DAG arguments

nikki-t committed:
Update CWL DAG arguments to remove ecr and add in log levels

nikki-t committed:
Add in SBG CWL and update modular DAG arguments

nikki-t committed:
Update CWL DAG arguments to remove ecr and add in log levels

nikki-t committed:
Add in new SBG test cases

nikki-t committed:
Increase SBG ISOFIT EC2 resources

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Temp hold

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Add in modular test for EMIT

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Change EC2 instance type for SBG E2E to troubleshoot potential spot issue

nikki-t commented:
The tests appear to be working in the dev environment after I changed the EC2 instance type for the SBG E2E test: https://github.com/unity-sds/unity-sps/actions/runs/15211329454/job/42785955045

It seemed like it kept getting terminated and I am wondering if that had something to do with the spot instance architecture.

The test environment failed at stage in: 

```bash
[2025-05-23T13:55:22.834+0000] {pod_manager.py:471} INFO - [base] 2025-05-23 13:55:22,756 [INFO] [botocore.credentials::1075] Found credentials from IAM Role: unity-test-sps-eks-eks-node-role
[2025-05-23T13:55:23.001+0000] {pod_manager.py:471} INFO - [base] 2025-05-23 13:55:22,833 [INFO] [botocore.credentials::1075] Found credentials from IAM Role: unity-test-sps-eks-eks-node-role
[2025-05-23T13:55:23.002+0000] {pod_manager.py:471} INFO - [base] 2025-05-23 13:55:22,999 [ERROR] [mdps_ds_lib.stage_in_out.download_granules_abstract::63] error downloading granule: urn:nasa:unity:unity:dev:SBG-L1B_PRE___1:SISTER_EMIT_L1B_RDN_20240103T131936_001
...
[2025-05-23T13:55:26.010+0000] {pod_manager.py:471} INFO - [base] botocore.exceptions.ClientError: An error occurred (403) when calling the HeadObject operation: Forbidden
```

It seems like a permissions issue?

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t commented:
Please note the OGC tests are commented out of the integration tests GitHub workflow: https://github.com/unity-sds/unity-sps/blob/develop/.github/workflows/integration_tests.yml which I think is due to integrating Cognito tokens into the OGC API tests infrastructure.

LucaCinquini reviewed (changes requested):
Hi @nikki-t , could you please change the integration tests so that the CWL Modular DAG can be run selectively on the Dev or Test venue? This way, we can have the CWL modular DAG tests succeed on the Dev venue, but not run at all on the Test venue.

The CWL classic DAG does so by adding an additional "venue" key to the configuration dictionary: 
[job_args_stage_out](https://github.com/unity-sds/unity-sps/blob/dad1f245ab1477555cd1a5b47cd59d1929903b92/unity-test/system/integration/step_defs/test_cwl_workflows_with_airflow_api.py#L98)

This way we should be able to run integration tests on this branch to completion (bypassing the Test venue for the CWL modular DAG).

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Conditionally run modular CWL DAG on dev venue only

nikki-t committed:
Fix formatting of integration tests

LucaCinquini committed:
Commenting out test for SBG scale

nikki-t committed:
Merge branch 'develop' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t committed:
Merge branch '394-increase-test-coverage' of github.com:unity-sds/unity-sps into 394-increase-test-coverage

nikki-t commented:
@LucaCinquini - I updated the CWL modular DAG tests so that they can be selectively run on test and have them executing on dev by default.

I updated this branch and ran the integration tests which passed: https://github.com/unity-sds/unity-sps/actions/runs/15535910428

LucaCinquini reviewed (approved):
Thanks @nikki-t , I will approve and merge.
